stdlib/2/__builtin__.pyi

```diff
@@ -242,11 +245,12 @@ class float:
     def __getnewargs__(self) -> Tuple[float]: ...
     if sys.version_info >= (3,):
         @overload
-        def __round__(self) -> int: ...
-        @overload
-        def __round__(self, ndigits: None) -> int: ...
+        def __round__(self, ndigits: None = ...) -> int: ...
         @overload
         def __round__(self, ndigits: int) -> float: ...
+        def __trunc__(self) -> int: ...
+        def __floor__(self) -> int: ...
+        def __ceil__(self) -> int: ...
```

Comment:
The last two don't seem to exist (which surprised me):

```python
Python 3.7.3 (default, Jun  3 2019, 18:27:33) 
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> x = 3.0
>>> x.__trunc__()
3
>>> x.__floor__()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'float' object has no attribute '__floor__'
>>> x.__ceil__()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'float' object has no attribute '__ceil__'
```

Reply:
Hmm, should have checked all of them.

I looked into cpython code, and what I understand is that `math.floor` (same for `math.ceil`) has a special case for floats, or rather types that can be converted to float (implement `__float__()`). In these cases, if `__floor__` is not defined, it converts to float and does the operation directly.

```py
import math

class X:
    def __float__(self):
        return 5.4

    #def __floor__(self):
    #    return 10

print(math.floor(X()))
```

Commented it prints 5, uncommented it prints 10.

So I think that, given `numbers.Real` already requires `__float__`, we can just diverge from the actual definition of `numbers.Real` and remove `__floor__` and `__ceil__` from it in typeshed. WDYT? [The `numbers.Real` doc](https://docs.python.org/3/library/numbers.html#numbers.Real) only requires `math.floor` to work.